ingestion-beam/src/main/java/com/mozilla/telemetry/schemas/SchemaStore.java

```diff
@@ -97,13 +105,15 @@ int numLoadedSchemas() {
   private void loadAllSchemas() throws IOException {
     final Map<String, T> tempSchemas = new HashMap<>();
     final Set<String> tempDirs = new HashSet<>();
+    Multimap<String, String> schemaAliasesMap;
     InputStream inputStream;
     try {
+      schemaAliasesMap = loadSchemaAliasesMap();
       Metadata metadata = FileSystems.matchSingleFileSpec(schemasLocation.get());
       ReadableByteChannel channel = FileSystems.open(metadata.resourceId());
       inputStream = Channels.newInputStream(channel);
     } catch (IOException e) {
-      throw new IOException("Exception thrown while fetching from configured schemasLocation", e);
+      throw new IOException("Exception thrown while loading schemas configuration", e);
```

Comment:
I'd prefer to keep schemaAliasesLocation as separated as possible from schemasLocation rather than nesting the loading of aliases inside loadAllSchemas.

Can we provide a loadSchemaAliases() method that gets called first and then either pass in the value to this method or have that method set a private variable to store the aliases?

That will separate concerns a bit more and allow us to keep these exception messages as specific as possible.

Reply:
I went one step further and decoupled loading aliases from loading schemas. This way it's not mixed with archive-reading code and would be easy to change/remove in the future.